esmvaltool/diag_scripts/ocean/diagnostic_timeseries.py

```diff
@@ -105,16 +160,18 @@ def make_time_series_plots(
 
         # Add title, legend to plots
         title = ' '.join([metadata['dataset'], metadata['long_name']])
-        if layer:
-            title = ' '.join(
-                [title, '(', layer,
-                 str(cube_layer.coords('depth')[0].units), ')'])
+        if layer != '':
+            if len(cube_layer.coords('depth')):
```

Comment:
the check to see if a `list` is empty is done `if list:` or `if not list:` rather than on its length as @bouweandela pointed out